jobsdb/jobsdb.go

```diff
@@ -2031,9 +2043,24 @@ func (jd *HandleT) WithStoreSafeTx(f func(tx StoreSafeTx) error) error {
 
 func (jd *HandleT) inStoreSafeCtx(f func() error) error {
 	// Only locks the list
-	jd.dsListLock.RLock()
-	defer jd.dsListLock.RUnlock()
-	return f()
+	op := func() error {
+		jd.dsListLock.RLock()
+		defer jd.dsListLock.RUnlock()
+		return f()
+	}
+	var err error
+	for {
+		err = op()
+		if err != nil && errors.Is(err, errStaleDsList) {
+			jd.logger.Errorf("[JobsDB] :: Store failed: %v. Retrying after refreshing DS cache", errStaleDsList)
+			jd.dsListLock.WithLock(func(l lock.DSListLockToken) {
+				_ = jd.refreshDSList(l)
+			})
+		} else {
+			break
+		}
+	}
+	return err
 }
```

Comment:
minor simplification
```suggestion
	for {
		err := op()
		if err != nil && errors.Is(err, errStaleDsList) {
			jd.logger.Errorf("[JobsDB] :: Store failed: %v. Retrying after refreshing DS cache", errStaleDsList)
			jd.dsListLock.WithLock(func(l lock.DSListLockToken) {
				_ = jd.refreshDSList(l)
			})
		} else {
			return err
		}
	}
}
```

Reply:
modified.